nengo/tests/test_connection.py

```diff
@@ -687,22 +687,35 @@ class Test(object):
         inst.sp = 'a'
 
 
-def test_nonexistant_prepost(Simulator):
+def test_prepost_errors(Simulator):
     with nengo.Network():
-        a = nengo.Ensemble(100, 1)
+        x = nengo.Ensemble(10, 1)
+        y = nengo.Ensemble(10, 1)
+        c = nengo.Connection(x, y)
+        with pytest.raises(ValueError):
+            nengo.Connection(3, y)
+        with pytest.raises(ValueError):
+            nengo.Connection(x, c)
```

Comment:
Yeah, always a question of how many things you want to test. I could also test connecting to a not-NengoObject above, and many other things. In this case, connecting to and from are running through the same validation code, so it's checked either way. I mean, someone could still mess it up, but they'd almost have to be trying. But if you're worried about that case for some reason, I could add it. Also, my general philosophy when fixing a bug is to add minimal tests to make sure that we don't revert, and cover the added code, since by that point the bug is already identified so the tests are less useful. For example, I'd rather add an additional test making sure we can't connect to Probes or some other NengoObject that's not connectable than beat a dead horse with the bug we've already solved.